docs/content/src/learn/tutorials/messaging-integrations/messaging-with-kafka/README.md

```diff
@@ -0,0 +1,272 @@
+};
+```
+
+We make use of a partitionAssignmentStrategy to assign the consumer to a particular topic partition.
+
+Let us now see the complete implementation of a consumer.
+
+#### consumer1.bal
+```ballerina
+import ballerina/kafka;
+import ballerina/io;
+import ballerina/lang.'string as strings;
+
+// Kafka Consumer Configuration
+kafka:ConsumerConfig consumer1 = {
+    bootstrapServers: "localhost:9092",
+    groupId: "consumer",
+    topics: ["product-price"],
+    pollingIntervalInMillis: 1000,
+    partitionAssignmentStrategy: "org.apache.kafka.clients.consumer.RoundRobinAssignor"
+};
+
+// Kafka Listener
+listener kafka:Consumer productConsumer1 = new (consumer1);
+
+// Service that listens to the particular topic
+service productConsumerService1 on productConsumer1 {
+    // Trigger whenever a message is added to the subscribed topic
+    resource function onMessage(kafka:Consumer productConsumer, kafka:ConsumerRecord[] records) returns error? {
+        foreach var entry in records {
+            byte[] serializedMessage = entry.value;
+            string|error stringMessage = strings:fromBytes(serializedMessage);
+
+            if (stringMessage is string) {
+                io:StringReader sr = new (stringMessage);
+                json jsonMessage = check sr.readJson();
+
+                io:println("ProductConsumerService1 : Product Received");
+                io:println("Name : ", jsonMessage.Name);
+                io:println("Price : ", jsonMessage.Price);
+            }
+        }
+    }
+}
+```
+
+In the above code, resource function `onMessage` will be triggered whenever a message published to the topic specified.
+
+
+## Deployment
+
+Once you are done with the development, you can deploy the services using any of the methods listed below.
+
+### Deploying locally
+
+To deploy locally, navigate to the `product_management_system` folder and execute the following command.
+
+```bash
+  $ ballerina build -a
+```
+This builds all the necessary bal files which produces a product_management_system.jar file.
+
+To run the service, navigate to the target/bin folder and run the following command
+
+```bash
+  $ java -jar product_management_system.jar
+```
+
+## Testing
+
+### Invoking the service
+
+- First, start the `ZooKeeper` instance with the default configurations by entering the following command in a terminal from `<KAFKA_HOME_DIRECTORY>`.
+
+ ```bash
+    $ bin/zookeeper-server-start.sh -daemon config/zookeeper.properties
+ ```
+
+(* `-daemon` flag is optional, use this if you want to run kafka server as a daemon) 
+
+  Here we start a zookeeper with default configurations (on `port:2181`).
+
+- Start a single `Kafka broker` instance with the default configurations by entering the following command  in a terminal from `<KAFKA_HOME_DIRECTORY>`.
+
+```bash
+   $ bin/kafka-server-start.sh -daemon config/server.properties
+```
+
+(* `-daemon` flag is optional, use this if you want to run kafka server as a daemon) 
```

Comment:
Change this to match the similar sentence above. Ensure that Kafka is written in titlecase.